doc/osmp_test_architecture.md

```diff
@@ -0,0 +1,224 @@
+Additional test can be performed on the SensorViewConfigRequest during the initialization of the model FMU, if implemented.
+
+## CL 3: Quantitative Verification
+
+For credibility assessment level 3, the model is again employed in a co-simulation with a trace file player or scenario engine.
+During the simulation, a trace file is generated containing the model output of every simulation time step.
+This is done by connecting a [trace file writer](https://github.com/Persival-GmbH/osi-sensordata-trace-file-writer) to the model output in the SSP specification of the test.
+The trace file writer is pre-built in every test environment.
+The FMU is located at _/tmp/tracefile_writer_fmu/osi-sensordata-trace-file-writer.fmu_.
+After the co-simulation is finished, the GitHub action scans the test folder for python files.
+If it finds one, the python script is called with the path to the just generated trace file as an argument.
+The file name of the python file does not matter.
+However, if there is more than one python file in the directory, the analysis will not be performed since it is unclear, which file to use.
+If supplementary files are necessary for the analysis, they are to be placed in a subfolder.
+
+The purpose is to test the proper implementation of the model code in terms of its quantifiable requirements.
+E.g. the sensor is rotated around all 3 axis, and it is evaluated, if the coordinate transformations in the model are correct.
+This can be achieved in two ways:
+
+1. The simulation trace file is analyzed by a custom standalone python script.
+2. The simulation trace file is compared to an expectation trace file with a python script.
+
+The general setup of this test level is depicted in the following image and both approaches are described in more detail in the following sections.
+
+<p align="center">
+  <img alt="CL 3 Test" src="./img/cl3_test.svg" width="600"> <br>
+  <em>Test setup for credibility assessment level 3</em>
+</p>
+
+### CL 3.1 Standalone Trace File Analysis
+
+The OSI trace file generated by the co-simulation is read in by a python script after the simulation has finished.
+The python script can perform a variety of tests on the trace file, depending on the specific scenario and use-case.
+For example, the scenario places an object at a certain distance inside the field of view of a sensor.
+The python script can analyze the sensor model output and check, if the object was detected at the expected distance.
+Another example would be to place an object outside of the field of view and check, that it was not detected.
+Furthermore, distribution functions for detections or expected number of detections on an object can be checked.
+
+An example implementation of this test can be found in the [sensor model template repository](https://github.com/openMSL/sl-1-0-sensor-model-repository-template/tree/main/test/integration/004_tracefile_analysis).
+
+### CL 3.2 Trace File Comparison
+
+(not yet implemented) <br>
```

Comment:
put note in the headline same as above?

Reply:
I would not put it in the headline, because the link to the section contains the section headline. This ways, links don't break, if we remove the (not yet implemented) in the future.

Reply:
Remove it above.